pkg/api/store/clustertemplate/store.go

```diff
@@ -209,6 +222,53 @@ func (p *Store) canSetEnforce(apiContext *types.APIContext, data map[string]inte
 			return httperror.NewAPIError(httperror.PermissionDenied, fmt.Sprintf("ClusterTemplate's %v field cannot be changed", managementv3.ClusterTemplateFieldEnforced))
 		}
 	}
+	return nil
+}
+
+func (p *Store) checkPermissionToCreateRevision(apiContext *types.APIContext, data map[string]interface{}) error {
+	userID := apiContext.Request.Header.Get("Impersonate-User")
+	if userID == "" {
+		return httperror.NewAPIError(httperror.NotFound, "invalid request: userID not found")
+	}
 
+	value, found := values.GetValue(data, managementv3.ClusterTemplateRevisionFieldClusterTemplateID)
+	if !found {
+		return httperror.NewAPIError(httperror.NotFound, "invalid request: clusterTemplateID not found")
+	}
+
+	clusterTemplateID := convert.ToString(value)
+	_, clusterTemplateName := ref.Parse(clusterTemplateID)
+
+	// check if rolebindings of type owner or member exist for this user for this clustertemplate
+	// if yes, then user is allowed to create revision for this template
+	// else not allowed since, either user has no access or only read-only access by either:
+	// 1. being added explicitly as read-only member OR
+	// 2. template is public, which gives everyone read-only access
+	canUpdate := false
+	for _, accessType := range []string{rrbac.OwnerAccess, rrbac.MemberAccess} {
+		rbName, _ := rrbac.GetRoleNameAndVerbs(accessType, clusterTemplateName, rrbac.ClusterTemplateResource)
+		// check if rolebinding with this name exists and has the user (caller of this request) as a subject
+		rb, err := p.rbLister.Get(namespace.GlobalNamespace, rbName)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
```

Comment:
Use httperror.WrapAPIError before returning the err?

Reply:
This will be k8s server error and unwrapped errors will throw 500